zulipterminal/ui_tools/views.py

```diff
@@ -1267,7 +1269,18 @@ def __init__(self, controller: Any, msg: Message, title: str,
         #               + 1 '<Any> Links' category label.
         slice_index = len(msg_info[0][1]) + 2
 
+        if topic_links:
+            topic_link_widgets, topic_link_width = (
+                self.create_link_buttons(controller, topic_links)
+            )
+
+            widgets = (widgets[:slice_index] + topic_link_widgets
+                       + widgets[slice_index:])
+            popup_width = max(popup_width, topic_link_width)
+
         if message_links:
+            if topic_links:
+                slice_index += (len(topic_links) + 2)
```

Comment:
Is this simplified if the slice_index is in the conditional in the earlier commit?